### PR TITLE
10132 New location events

### PR DESCRIPTION
**Pull request by Sergio-Mira**
https://swrvedev.jira.com/browse/SWRVE-10132

### CODE REVIEW

File: Sdk/Track/Swrve.m [hunk 1]
```diff
@@ -1376,6 +1376,24 @@ -(int) eventInternal:(NSString*)eventName payload:(NSDictionary*)eventPayload tr
     return SWRVE_SUCCESS;
 }
 
+-(int) locationImpressionEvent:(int)messageId
+{
+    [self maybeFlushToDisk];
+    NSMutableDictionary* json = [[NSMutableDictionary alloc] init];
+    [json setValue:[[NSNumber alloc] initWithInt:messageId] forKey:@"id"];
+    [self queueEvent:@"location_campaign_impression" data:json triggerCallback:NO];
```
Comment (dominicmarmionswrve) [inline Sdk/Track/Swrve.m]: Can we move this method to the locationsdk and just call queueEvent?

File: Sdk/Track/Swrve.m [hunk 2]
```diff
@@ -1376,6 +1376,24 @@ -(int) eventInternal:(NSString*)eventName payload:(NSDictionary*)eventPayload tr
...
+    return SWRVE_SUCCESS;
+}
+
+-(int) locationEngagedEvent:(int)messageId
```
Comment (dominicmarmionswrve) [inline Sdk/Track/Swrve.m]: And this method too.

File: Sdk/Track/Swrve.m (hunk 1 above)
Comment (Sergio-Mira) [inline Sdk/Track/Swrve.m]: Yeah, I can do that

**Comment by dominicmarmionswrve** [comment]
:rocket:

**Pull request merged**